src/core_modules/capture-core/components/Pages/MainPage/WorkingListsCommon/hooks/useWorkingListsCommonStateManagement.js

```diff
@@ -25,103 +25,151 @@ import {
+    CancelLoadView,
+    CancelUpdateList,
+    ChangePage,
+    ChangeRowsPerPage,
+    ClearFilter,
+    LoadTemplates,
+    LoadView,
+    SelectRestMenuItem,
+    SelectTemplate,
+    SetColumnOrder,
+    Sort,
+    UpdateFilter,
+    UpdateList,
+} from '../../WorkingLists';
+import type { AddTemplate, DeleteTemplate, UpdateTemplate } from '../../WorkingListsCommon';
 
-const useTemplates = (listId: string, workingListDispatch: Function) => {
+const useTemplates = (
+    dispatch: ReduxDispatch,
+    { storeId, workingListsType }: { storeId: string, workingListsType: string }) => {
     const templateState = useSelector(({ workingListsTemplates }) => ({
-        currentTemplate: workingListsTemplates[listId] &&
-            workingListsTemplates[listId].selectedTemplateId &&
-            workingListsTemplates[listId].templates &&
-            workingListsTemplates[listId].templates.find(template => template.id === workingListsTemplates[listId].selectedTemplateId),
-        templates: workingListsTemplates[listId] &&
-            workingListsTemplates[listId].templates,
-        templatesLoading: !!workingListsTemplates[listId] &&
-            !!workingListsTemplates[listId].loading,
-        loadTemplatesError: workingListsTemplates[listId] && workingListsTemplates[listId].loadError,
+        currentTemplate: workingListsTemplates[storeId] &&
+            workingListsTemplates[storeId].selectedTemplateId &&
+            workingListsTemplates[storeId].templates &&
+            workingListsTemplates[storeId].templates.find(template => template.id === workingListsTemplates[storeId].selectedTemplateId),
+        templates: workingListsTemplates[storeId] &&
+            workingListsTemplates[storeId].templates,
+        templatesLoading: !!workingListsTemplates[storeId] &&
+            !!workingListsTemplates[storeId].loading,
+        loadTemplatesError: workingListsTemplates[storeId] && workingListsTemplates[storeId].loadError,
     }), shallowEqual);
 
     const templateDispatch = useMemo(() => ({
-        onSelectTemplate: workingListDispatch(selectTemplate),
-        onLoadTemplates: workingListDispatch(fetchTemplates),
-        onCancelLoadTemplates: workingListDispatch(fetchTemplatesCancel),
-        onAddTemplate: workingListDispatch(addTemplate, (name: string, criteria: Object, data: Object) => [
-            name,
-            criteria, {
-                ...data,
-                listId,
-            },
-        ]),
-        onUpdateTemplate: workingListDispatch(updateTemplate, (template: Object, criteria: Object, data: Object) => [
-            template,
-            criteria, {
-                ...data,
-                listId,
-            },
-        ]),
-        onDeleteTemplate: workingListDispatch(deleteTemplate),
-        onCleanSkipInitAddingTemplate: workingListDispatch(cleanSkipInitAddingTemplate),
-    }), [listId, workingListDispatch]);
+        onSelectTemplate: (...args) => dispatch(selectTemplate(...args, storeId)),
```

Comment:
Looking at it it looks to me that `selectTemplate` takes only one argument, `storeId`. Do we really need args here?